tests/serac_quasistatic_solver.cpp

```diff
@@ -136,17 +116,80 @@ TEST(nonlinear_solid_solver, qs_direct_solve)
   // Set the material parameters
   solid_solver.setHyperelasticMaterialParameters(0.25, 10.0);
 
-  // Set the linear solver params
-  serac::LinearSolverParameters params;
-  params.lin_solver = serac::LinearSolver::SuperLU;
+  // Initialize the output
+  solid_solver.initializeOutput(serac::OutputType::VisIt, "static_solid");
+
+  // Complete the solver setup
+  solid_solver.completeSetup();
+
+  double dt = 1.0;
+  solid_solver.advanceTimestep(dt);
+
+  solid_solver.outputState();
+
+  mfem::Vector zero(dim);
+  zero = 0.0;
+  mfem::VectorConstantCoefficient zerovec(zero);
+
+  double x_norm = solid_solver.displacement()->gridFunc().ComputeLpError(2.0, zerovec);
+
+  EXPECT_NEAR(2.2309025, x_norm, 0.001);
+
+  MPI_Barrier(MPI_COMM_WORLD);
+}
+*/
 
-  serac::NonlinearSolverParameters nl_params;
-  nl_params.rel_tol     = 1.0e-3;
-  nl_params.abs_tol     = 1.0e-6;
-  nl_params.print_level = 1;
-  nl_params.max_iter    = 5000;
+TEST(nonlinear_solid_solver, qs_custom_solve)
+{
+  MPI_Barrier(MPI_COMM_WORLD);
 
-  solid_solver.setSolverParameters(params, nl_params);
+  // Open the mesh
+  std::string mesh_file = std::string(SERAC_REPO_DIR) + "/data/beam-hex.mesh";
+
+  auto pmesh = buildMeshFromFile(mesh_file, 1, 0);
+
+  int dim = pmesh->Dimension();
+
+  // Simulate a custom solver by manually building the linear solver and passing it in
+  // The custom solver built here should be identical to what is internally built in the
+  // qs_solve test
+  auto custom_params = NonlinearSolid::default_qs_linear_params;
+  auto custom_solver = std::make_unique<mfem::MINRESSolver>(MPI_COMM_WORLD);
+  custom_solver->SetRelTol(custom_params.rel_tol);
+  custom_solver->SetAbsTol(custom_params.abs_tol);
+  custom_solver->SetMaxIter(custom_params.max_iter);
+  custom_solver->SetPrintLevel(custom_params.print_level);
+
+  // Define the solver object
+  NonlinearSolid solid_solver(
+      1, pmesh,
+      // Compiler cannot deduce through two layers of indirection so explicitly initialize the CustomParams here
+      // Note that a non-owning pointer is used and that the lifetime of the physics module is dependent on that
+      // of the driver
+      {CustomSolverParameters{custom_solver.get()}, NonlinearSolid::default_qs_nonlinear_params, std::nullopt});
```

Comment:
I think this is a bit to convoluted for user-facing code. A struct of parameters (or more likely an inlet object) and a solver pointer would be a cleaner interface.

Reply:
I agree - this has been changed to a struct:
```cpp
NonlinearSolid::NonlinearSolidParameters params;
params.H_lin_params    = CustomSolverParameters{custom_solver.get()};
params.H_nonlin_params = NonlinearSolid::default_qs_nonlinear_params;

// Define the solver object
NonlinearSolid solid_solver(1, pmesh, params);
```